src/main/js/components/header/tableSettings/TableSettings.jsx

```diff
@@ -0,0 +1,56 @@
+/** Displays a cogwheel icon that, when clicked, displays an instance of TableSettingsPopup */
+
+import React from "react";
+import TableSettingsPopup from "./TableSettingsPopup";
+import {contains} from "lodash/fp"
+
+class TableSettings extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {open: false};
+  }
+
+  setOpenState = open => {
+    console.log("setOpenState:", open);
+    this.setState({open: open})
+  };
+
+  toggleSettingsPopup = () => {
+    this.setOpenState(!this.state.open);
+  };
+
+  onClickOutside = evt => {
+    const target = evt.target;
+    const el = document.getElementById("table-settings");
```

Comment:
in this case it would be possible to use [react refs](https://facebook.github.io/react/docs/refs-and-the-dom.html)